ckanext/unaids/blueprints/unaids_dataset_transfer.py

```diff
@@ -0,0 +1,54 @@
+# encoding: utf-8
+from flask import Blueprint
+from ckan import model
+from ckan.plugins import toolkit
+from ckan.common import _, c
+import ckan.lib.helpers as h
+import ckanext.unaids.helpers as helpers
+import ckan.logic as logic
+import ckan.lib.base as base
+
+unaids_dataset_transfer = Blueprint(
+    u'unaids_dataset_transfer',
+    __name__,
+    url_prefix=u'/validation'
+)
+
+
+def process_dataset_transfer(dataset_id):
+    dataset = toolkit.get_action('package_show')({}, {'id': dataset_id})
+    
+    # validate user can transfer dataset
+    valid = helpers.check_organization_update_access(
+        dataset['org_to_allow_transfer_to']
```

Comment:
Writing our own authorisation logic is fine, but let's register it properly as an auth check and call it with check access. I think you need to do the following (this is broadly correct, but may not be 100% correct): 
 1) Define a function with all your auth logic.  It should take arguments "context" and "data_dict", being dictionaries of the form `{'user': '<user_object>'}` and  `{'org_to_allow_transfer_to': '<org_id>'}` respectively.  The function should return a dict: `{'success': True}` if it passes. 
 2) Make the unaids plugin implement the [IAuthFunctions interface](https://docs.ckan.org/en/2.9/extensions/plugin-interfaces.html#ckan.plugins.interfaces.IAuthFunctions).  This is done at the top of the UNAIDSPlugin class with the line `p.implements(p.IAuthFunctions)`
3) Add the function get_auth_functions to the UNAIDSPlugin class. The function should return a dictionary `{'organization_update_access': <auth fucntion from step 1>}` (A bit like registring a helper or action).
4) Then here in the process_dataset_transfer function you should call ```h.check_access('organization_update_access', {'org_to_allow_transfer_to': dataset['org_to_allow_transfer_to']})```  (h being ckan.lib.helpers)

Reply:
Great feedback. This is now done ✅